composer/callbacks/checkpoint_saver.py

```diff
@@ -382,7 +382,12 @@ def _save_checkpoint(self, state: State, logger: Logger):
                 ).lstrip('/')
 
             log.debug(f'Uploading checkpoint to {remote_file_name}')
-            logger.upload_file(remote_file_name=remote_file_name, file_path=saved_path, overwrite=self.overwrite)
+            try:
+                logger.upload_file(remote_file_name=remote_file_name, file_path=saved_path, overwrite=self.overwrite)
+            except FileExistsError as e:
+                raise FileExistsError(
+                    f'Uploading checkpoint failed with error: {e}. overwrite was set to {self.overwrite}. To overwrite checkpoints with Trainer, set save_overwrite to True.'
+                )
```

Comment:
@irenedea I think you are missing a `raise ... from e` here. As is, this will discard the stack trace. Can you open a new PR adding in the trace?